Katello/katello
# Fixes #27211 - Publish Pulp 3 Docker CV with Filters

Adds CV filter support for Pulp 3 CVs with Docker content.  I've also re-factored some of the logic in `copy_all_units.rb` to hopefully make it easier to understand, but let me know if anyone has better ideas for re-factoring that.

To test:
1) Spin up a Pulp 3 dev box: https://projects.theforeman.org/projects/katello/wiki/Pulp_3_Integration
2) Sync a repo with some Docker content
3) Publish a CV with some Docker content using both include and exclude filters
4) Ensure correct content count in Katello (keeping in mind that [this bug](https://projects.theforeman.org/issues/27882) might still be around)
5) Ensure correct content count in Pulp 3 by querying the new CV version repo's version_href using the Pulp 3 API

~~Note: to test this correctly, the API endpoint in the Pulp 3 recursive-add binding code needs to be updated to include /pulp/api/v3/.  Ensure `local_var_path = '/pulp/api/v3/docker/recursive-add/'` in `docker_recursive_add_api.rb`~~

~~Related Pulp 3 issue: https://pulp.plan.io/issues/5486~~
The above has been fixed by bumping the docker client version.
